youtube/yt_data_extract/common.py

```diff
@@ -343,6 +346,48 @@ def extract_item_info(item, additional_info={}):
         else:
             info['index'] = None
 
+    elif primary_type == 'short':
+        info['id'] = item.get('videoId')
+        if not info['id']:
+            info['id'] = deep_get(item,'navigationEndpoint',
+                                  'reelWatchEndpoint', 'videoId')
+        info['approx_view_count'] = extract_approx_int(item.get('viewCountText'))
+
+        # handle case where it is "No views"
+        if not info['approx_view_count']:
+            if ('No views' in item.get('shortViewCountText', '')
+                    or 'no views' in accessibility_label.lower()):
+                info['view_count'] = 0
+                info['approx_view_count'] = '0'
+
+        # dig into accessibility data to get duration for shorts
+        accessibility_label = multi_deep_get(item,
+            ['accessibility', 'accessibilityData', 'label'],
+            default='')
+
+        duration = re.search(r'(\d+) (second|seconds|minute) - play video',
+                             accessibility_label)
+        if duration.group(2) == 'minute':
+            info['duration'] = "1:00"
+        else:
+            info['duration'] = "0:" + duration.group(1).zfill(2)
```

Comment:
Need to handle the case where re.search returns None due to no match, otherwise an exception will occur
Additionally, it would be better to include these lines in the `if primary_type == 'video'` block in order to avoid repeating code. Run them as a backup if info['duration'] is None with a comment indicating its for shorts.

Reply:
Changes made. Initially I also wanted to put both scenarios under the video block, but decided against it after realizing the video id, views, and duration were all stored in slightly different places. After including them under the duration check, I can see it wasn't that big a difference.